tfjs-backend-webgl/src/pool_gpu.ts

```diff
@@ -342,7 +342,10 @@ export class Pool3DProgram implements GPGPUProgram {
     let returnValue = `${poolType}(${poolType}(${poolType}(` +
         'minMaxValue[0], minMaxValue[1]), minMaxValue[2]), minMaxValue[3])';
     if (poolType === 'avg') {
-      returnValue = `avgValue / count`;
+      // Use `max(count, 1.0)` instead of `count` in case count === 0.0.
+      // If count === 0.0, `avgValue` is always 0.0 and we change `count`'s
+      // value to avoid dividing zero.
+      returnValue = `avgValue / max(count, 1.0)`;
```

Comment:
cc @qjia7 @xhcao , WebGPU may have the same issue, https://github.com/tensorflow/tfjs/blob/43143347656765eb174a1712fbef96077b105a2e/tfjs-backend-webgpu/src/pool2d_webgpu.ts#L56

Reply:
Thanks @Linchenn . Just curious, is L124 `returnValue = `avgValue / count`;` missed for change? 

And another question is that it seems that it's not possible to happen that `count` is zero in https://github.com/tensorflow/tfjs/blob/43143347656765eb174a1712fbef96077b105a2e/tfjs-backend-webgpu/src/pool2d_webgpu.ts#L56 since `updateSnippet` will increase `count`. Will there be a situation that the filter window is totally no overlap with the input window?

Reply:
Great catch, thanks!

If padding >= filter size, this would happen, as https://github.com/tensorflow/tfjs/issues/7122.

Reply:
LGTM. Thanks for your explanation.
Can you help to cover the webgpu change in this PR since you already find the right place? :) And  it will be great if you add a similar case as #7122 to file `avg_pool_3d_test.ts`?

Reply:
Done. Thank you!